### PR TITLE
Preserve data Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
**Description**
For this final milestone of our portfolio website, We have saved the form data in the local storage of the browser. That way when the user reloads the page the data they filled out in the form will be preserved.

**Project_ requirements**

> We implemented the following interactions:

     - When the user changes the content of any input field, the data is saved to the local storage.
     - When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
     

> We used the following data model:

     - We created a single JavaScript object with all the data from the entire form and save it in local storage.

### CODE REVIEW

File: Js/preserve-data.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  const data = {
+    id: Date.now(),
+    name: document.getElementById('name').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
+  datas.push(data);
+  document.forms[0].reset();
+
+  console.warn('Adjected', { datas });
```
Comment (Olamarx) [inline Js/preserve-data.js:14]: [OPTIONAL] You probably forgot to remove this console

File: Js/preserve-data.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  localStorage.setItem('InputDataList', JSON.stringify(datas));
```
Comment (Olamarx) [inline Js/preserve-data.js:16]: - [ ] I can see you are setting the item to the local storage upon submitting the form. This should not be the case, we should not submit the form before setting to the localStorage, it should be on the event of input change of either of the three inputs and we set the data to local storage. Also, we should getItem from localStorsge if InputDataList is true in the local storage upon loading the webpage ` let savedFormData = localStorage.getItem('contactFormInput');
savedFormData = JSON.parse(savedFormData);`
. Kindly have a closer check. Tag me if it unclear, please.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @HERMON-1995, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️ 
- SetItem to Local Storage ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly fix the MIT license link on your readme file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Js/preserve-data.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const addData = (ev) => {
+  ev.preventDefault();
+  const data = {
+    id: Date.now(),
+    name: document.getElementById('name').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
+  datas.push(data);
+  document.forms[0].reset();
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply Js/preserve-data.js]: Fixed.

File: Js/preserve-data.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply Js/preserve-data.js]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @HERMON-1995 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git Hub-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**